### ISSUE TITLE
How to communicate with mongo from another container

### ISSUE THREAD

**Issue by DaniBet** [containers/podman#9539]
Hello,

I have two pods developed with kube in podman. My issue is that I don't know how to comunicate with the mongo pod from the another one. 
In kubernetes mongo has a link something like : mongo.namespace.svc.cluster-local. How can I do that in podman?

**Comment by rhatdan** [comment]
@mheon @haircommander @baude PTAL, Could we handle this with the dnsname plugin.

**Comment by mheon** [comment]
Sure, via aliases, but we'd need someone to tell us what aliases to generate. I get the feeling the logic for figuring out what specific services get called is not trivial.

**Comment by rhatdan** [comment]
@umohnani8 @haircommander WDYT?

**Comment by haircommander** [comment]
I believe this is what services are for, which podman {generate,play} should support

**Comment by FilBot3** [comment]
Just bumping again. 

I do find myself frequently thinking about using Kube's `kind: Service` to expose a Pod like a load balancer. Could a Podman specific annotation or label for the `kind: Pod` or `kind: Deployment` be used as a Podman network specific DNS?

**Comment by rhatdan** [comment]
@Luap99 @mheon @baude WDYT?

**Comment by Luap99** [comment]
The pod name can already be used for dns resolution. The problem is that for default network  dns is not enabled. If you create a new network with `podman network create mynet` and start your pods with `podman play kube --network mynet ...` the pods should be able to communicate via pod name.

**Comment by FilBot3** [comment]
So somethign similar to

In shell 1

```bash
podman network create phil-test
```

In shell 2

```bash
podman run --rm -it --network=phil-test -p 4444 --name=ubuntu1 docker.io/library/ubuntu:20.04 /bin/bash

$ apt update && apt install netcat -y
$ netcat -l 4444
```

In shell 3

```bash
podman run --rm -it --network=phil-test --name=ubuntu2 docker.io/library/ubuntu:20.04 /bin/bash

$ apt udpate && apt install netcat -y
$ netcat -vvz ubuntu1 4444
```

it resolved

**Comment by rhatdan** [comment]
You should be able to setup the "phil-test" to be your default network as well in containers.conf.

**Comment by FilBot3** [comment]
@DaniBet , does this help OP solve their question?

**Comment by DaniBet** [comment]
In my point of view, this approach is not the normal way, you should have a generic way of addressing. If I have a Podman file where I have 3 containers, I can not use the name of the container in order to reach it, like in Docker or Kubernetes.

**Comment by FilBot3** [comment]
> In my point of view, this approach is not the normal way, you should have a generic way of addressing. If I have a Podman file where I have 3 containers, I can not use the name of the container in order to reach it, like in Docker or Kubernetes.

Doing it this way was how I did it with Docker. Using Docker-Compose, it automatically sets this up for you and does all the networking and container naming. 

What do you mean by `Podman file`?

**Comment by DaniBet** [comment]
With Podman file I mean the kube file where I describe the containers. In the normal way, I should have a way to address to the other containers. 
If I have two containers, container x and y, and in the second container I need to point something to the first, I can not do it. In the normal way, in my point of view, using the name of the container   like in kubernetes or docker.

**Comment by mheon** [comment]
This works if you use any network that is not the default one. We do this for compatibility with Docker, who do not enable DNS on their default network. If you really want to, you can edit the default network we provide and add the following config snippet to enable DNS aliasing:

```
            {
                "capabilities": {
                    "aliases": true
                },
                "domainName": "dns.podman",
                "type": "dnsname"
            }
```

@baude @Luap99 @rhatdan Does restricting DNS on the default network actually make sense? Are there use-cases that actually rely on this *not* working?

**Comment by Luap99** [comment]
@mheon Yes, dns should not be on by default (at least not the dnsname plugin). The problem is that dnsname binds port 53 on the bridge interface and therefore can conflict with other dns servers on the host. I also think the extra dnsmasq process will bother people. Lastly, the default docker network also does not have dns support.

**Comment by mheon** [comment]
We may wish to reconsider this after we rewrite the DNS bits, if we have a persistent server already and don't need to bind to a localhost address in use elsewhere, it seems like it might make sense to flip the default. Performance could be another concern, though, would need to verify it's (at least almost) as fast as not having it.

**Comment by rhatdan** [comment]
Yes in a perfect world this would work out of the box.

**Comment by baude** [comment]
you can also delete the default network and create a new one and that will automatically add it

**Comment by rhatdan** [comment]
@Luap99 @baude @mheon Is this something we hope to have a fix for in 4.0?

**Comment by mheon** [comment]
Maybe? @Luap99 when we move the CNI DNS plugin into Podman, we can make it default (at least for Netavark) and resolve this.

**Comment by baude** [comment]
As was said earlier, the behavior of podman follows docker on the default network ... and for a good reason.  What is tricking people here is that compose always makes a new network, which is why the name resolution works.  If *any* change is made, we should consider either doing the same with play kube or having a command line switch for this.  @rhatdan wdyt?

**Comment by rhatdan** [comment]
I think we should do it by default and match docker-compose.

**Comment by FilBot3** [comment]
To clarify, you're saying that Podman should match what `docker-compose`/`podman-compose` do in creating a network with DNS enabled so that Pods can communicate with each other?

**Comment by rhatdan** [comment]
Yes I think that makes sense.  If I create multiple Pods within the same kubernetes.YAML, would it be assumed that they can communicate?

**Comment by baude** [comment]
this is verified to work with podman 4.x and the netavark/aardvark-dns network backend.  you can do this simply with standard tcp/ip with play kube.  the one gotcha is that the podman's default network does not have dns enabled, so you need to create a new network (`sudo podman network create new1`) and then ensure you also use the `--network` option in play kube.  For example:

```
$ sudo podman play kube --network new1 mykube.yaml
....
